### DISCUSSION TITLE
List groups Apicurio Registry API

### DISCUSSION THREAD

**Discussion by cesareomacias** [Apicurio/apicurio-registry#5214] [Q&A]
Hi everyone!

I have the sr-admin role (OIDC), and when I make this request:

`https://{myHost}/apis/registry/v2/groups`

I expect to retrieve all groups available in the registry, but I'm retrieving only the groups which I'm owner (I guess):
```
{
	"count": 2,
	"groups": [
		{
			"createdBy": "e535c0******e852365",
			"createdOn": "2024-09-19T14:48:35+0000",
			"id": "org1",
			"modifiedBy": "e535c0******e852365",
			"modifiedOn": "2024-09-19T14:48:35+0000"
		},
		{
			"createdBy": "{myUser}",
			"createdOn": "2024-07-31T09:30:16+0000",
			"id": "org2",
			"modifiedBy": "{myUser}",
			"modifiedOn": "2024-07-31T09:30:16+0000"
		}
	]
}
```

Is that the expected behavior? How I could retrieve the whole list of groups?

Regards!

**Comment by carlesarnal** [comment]
Hello @cesareomacias, the expectation is to get all the groups. That said, do you mind adding a bit more information, like which Registry version you're using etc? Thanks

**Reply by cesareomacias** [reply]
Hi @carlesarnal 

Sure, I deployed 2.6.3.Final version into an AWS EC2 Instance. it's a production environment, so I have a sql db in order to persist the artifacts.

Thanks!

**Reply by EricWittmann** [reply]
In version 2.6 you can't filter by owner even if you wanted to.

**Reply by cesareomacias** [reply]
Hi @EricWittmann My goal is not filter by Owners, just retrieve all groups.

I've downloaded all artifacts and it's interesting because looking into the groups folder, I found:
![image](https://github.com/user-attachments/assets/c5300b5f-f663-4b26-a894-f098a4c916ad)

It's just retrieving the *.Group.json "groups/files" not the folders (real groups).

Just for context, I also have another instance running 2.2.5.Final, then I exported all artifacts and imported in the 2.6.3.Final instance.

Regards.

**Reply by EricWittmann** [reply]
Okay. I think what's happening here is that you are running a version of registry that does not support groups with metadata. In other words registry used to have groups only as a piece of metadata on the artifact.

More recent versions formalized the group concept to be a real entity complete with metadata. 

If you export the content from an older version of registry it will not include the group information. Because there is no group information. 

Probably we should now be automatically creating groups when importing content. But we do not currently do that.

@carlesarnal what do you think? Should we call Ensure Group when importing?  We could make this change for 3.0.

**Reply by cesareomacias** [reply]
Hi @EricWittmann @carlesarnal 

OK, got it. Finally, that change will be available in 3.x.x or do you think is possible to apply in 2.6.X ?

Regards.

**Reply by carlesarnal** [reply]
Interesting. For 2.6.x I would expect this to be handled automatically by the upgrade process. There are also other things happening that are very important (like hash recalculation) that would be missing otherwise. I would apply that change to 3.0, since using the export feature there makes sense, but not to 2.6.x, since the flow there should be, IMO, first upgrade the server with the data (creating a backup first, of course) and then migrate to the new one.

**Reply by EricWittmann** [reply]
Added an issue for 3.x here:
* https://github.com/Apicurio/apicurio-registry/issues/5401

**Comment by EricWittmann** [comment]
This was determined to be a bug, with a fix in this PR:

* https://github.com/Apicurio/apicurio-registry/pull/5476
